module-system/sov-modules-macros/tests/custom_codec_must_be_used.rs

```diff
@@ -0,0 +1,46 @@
+use std::panic::catch_unwind;
+
+use sov_modules_api::default_context::ZkDefaultContext;
+use sov_modules_api::{Context, ModuleInfo};
+use sov_state::{DefaultStorageSpec, ProverStorage, StateValue, StateValueCodec, WorkingSet};
+
+#[derive(ModuleInfo)]
+struct TestModule<C>
+where
+    C: Context,
+{
+    #[address]
+    pub address: C::Address,
+
+    #[state(codec_builder = "Default::default")]
+    pub state_value: StateValue<u32, CustomCodec>,
```

Comment:
Are we getting a user-friendly error of `CustomCodec` doesn't implement `StateValueCodec`? Let's add a test for it.

Reply:
Good thinking, I'll do that

Reply:
Unfortunately we're not in control of type bounds -related error messages within the proc macro. I have modified the type bounds to imporve the error message, though: https://github.com/Sovereign-Labs/sovereign-sdk/pull/648/commits/11f13dc0a8bcd8b08a791689f39d447ea57e534f